discovery/syncer.go

```diff
@@ -894,23 +903,22 @@ func (g *GossipSyncer) replyShortChanIDs(query *lnwire.QueryShortChanIDs) error
 			query.ShortChanIDs[0].ToUint64(), err)
 	}
 
-	// If we didn't find any messages related to those channel ID's, then
-	// we'll send over a reply marking the end of our response, and exit
-	// early.
-	if len(replyMsgs) == 0 {
-		return g.cfg.sendToPeer(&lnwire.ReplyShortChanIDsEnd{
-			ChainHash: query.ChainHash,
-			Complete:  1,
-		})
+	// Reply with any messages related to those channel ID's, we'll write
+	// each one individually and synchronously to throttle the sends and
+	// perform buffering of responses in the syncer as opposed to the peer.
+	for _, msg := range replyMsgs {
+		err := g.cfg.sendToPeerSync(msg)
```

Comment:
This should help with some of the write timeout issues as well since it should move more or less at the pace of their processing.

Reply:
that's the hope! there's still a case which cannot be handled by this, which the write+flush fix solves completely. mainly that even if these sends are throttled, it can still get jammed if the remote peer is slow to process after reading and we try to send one more message. with this, we are less likely to hit that case, but it will go away completely with #2924